gossip/src/duplicate_shred.rs

```diff
@@ -90,6 +90,22 @@ pub enum Error {
     UnknownSlotLeader(Slot),
 }
 
+impl Error {
+    /// Errors indicating that the initial node submitted an invalid duplicate proof case
+    pub(crate) fn is_non_critical(&self) -> bool {
+        matches!(
```

Comment:
I would rather we explicitly list every enum variant and what is critical and what is not, similar to:
https://rust-lang.github.io/rust-clippy/master/index.html#/wildcard_enum_match_arm
but I guess not much an issue here.